src/preamble.js

```diff
@@ -134,16 +134,194 @@ var HEAP,
 var HEAP_DATA_VIEW;
 #endif
 
+function fixPointer (ptr) {
+  if (typeof ptr === 'number' && ptr < 0) {
+    return ptr >>> 0;
+  }
+  return ptr;
+}
+
+#if MEMORY64
+/**
+ * @param {string} type - Heap type
+ * @param {number} [offset] - Heap offset
+ * @param {number} [length] - typed array length
+ */
+const getHeapBlock =  (type, offset, length) => {
```

Comment:
Why not define this helper function inside of `updateMemoryViews` along with the all the ones?